src/shared/components/common/markdown-textarea.tsx

```diff
@@ -374,10 +374,59 @@ export class MarkdownTextArea extends Component<
     autosize.update(textarea);
   }
 
-  handleImageUploadPaste(i: MarkdownTextArea, event: any) {
+  handlePaste(i: MarkdownTextArea, event: ClipboardEvent) {
+    if (!event.clipboardData) return;
+
+    // check clipboard files
     const image = event.clipboardData.files[0];
     if (image) {
       i.handleImageUpload(i, image);
+      return;
+    }
+
+    // check clipboard url
+    const url = i.isValidUrl(event.clipboardData.getData("text"));
+    if (url) {
+      i.handleUrlPaste(url, i, event);
+      return;
+    }
+  }
+
+  handleUrlPaste(url: URL, i: MarkdownTextArea, event: ClipboardEvent) {
+    // query textarea element
+    const textarea = document.getElementById(i.id);
+
+    if (textarea instanceof HTMLTextAreaElement) {
+      event.preventDefault();
+      const { selectionStart, selectionEnd } = textarea;
+      const selectedText = i.getSelectedText() || url.toString();
+
+      // update textarea content
+      i.setState({
+        content: `${
+          i.state.content ? i.state.content.substring(0, selectionStart) : ""
+        }[${selectedText}](${url.toString()})${
+          i.state.content ? i.state.content.substring(selectionEnd) : ""
+        }`,
+      });
```

Comment:
I'm pretty sure inferno has similar issues to react when trying to set state using a previous value of that state. Try this:
```suggestion
     i.setState(({content}) => ({
         content: `${content?.substring(0, selectionStart) ?? ""}[${selectedText}](${url.toString()})${content?.substring(selectionEnd) ?? ""}`
     }));
```

You will probably need to format this with prettier still.